includes/class-newspack-popups-newsletters.php

```diff
@@ -153,60 +102,47 @@ private static function fetch_reader_data_from_esp( $email_address ) {
-			if ( ! is_wp_error( $subscribed_lists ) && ! empty( $subscribed_lists ) && is_array( $subscribed_lists ) ) {
-				// The reader is subscribed to one or more lists, so they should be segmented as a subscriber.
+		// Look up the email address as a contact with the connected ESP. If not a contact, no need to proceed.
+		$subscribed_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email_address );
+		if ( ! is_wp_error( $subscribed_lists ) && ! empty( $subscribed_lists ) && is_array( $subscribed_lists ) ) {
+			// The reader is subscribed to one or more lists, so they should be segmented as a subscriber.
+			$events_to_add[] = [
+				'type'    => 'subscription',
+				'context' => $email_address,
+				'value'   => [
+					'esp'   => $newsletter_provider->service,
+					'lists' => $subscribed_lists,
+				],
+			];
+
+			// TODO: set subscription status in the Reader Data store.
+		}
+
+		$contact_details = \Newspack_Newsletters_Subscription::get_contact_data( $email_address, true );
+		if ( ! is_wp_error( $contact_details ) && isset( $contact_details['metadata'] ) ) {
+			$metadata        = $contact_details['metadata'];
+			$donation_amount = 0;
+
+			if ( isset( $metadata['NP_LAST_PAYMENT_AMOUNT'] ) ) {
+				$donation_amount = (int) $metadata['NP_LAST_PAYMENT_AMOUNT'];
+			} elseif ( isset( $metadata['TOTAL_SPENT'] ) ) {
+				$donation_amount = (int) $metadata['TOTAL_SPENT'];
+			}
+			if ( 0 < $donation_amount ) {
 				$events_to_add[] = [
-					'type'    => 'subscription',
-					'context' => $email_address,
+					'type'    => 'donation',
+					'context' => $newsletter_provider->service,
 					'value'   => [
-						'esp'   => $newsletter_provider->service,
-						'lists' => $subscribed_lists,
+						'amount' => $donation_amount,
 					],
 				];
 			}
-		}
-
-		// If the reader is already known to be a donor, no need to proceed.
-		$has_donation_events = ! empty( $api->get_reader_events( $client_id, 'donation', $email_address ) );
-		if ( ! $has_donation_events ) {
-			$contact_details = \Newspack_Newsletters_Subscription::get_contact_data( $email_address, true );
-			if ( ! is_wp_error( $contact_details ) && isset( $contact_details['metadata'] ) ) {
-				$metadata        = $contact_details['metadata'];
-				$donation_amount = 0;
-
-				if ( isset( $metadata['NP_LAST_PAYMENT_AMOUNT'] ) ) {
-					$donation_amount = (int) $metadata['NP_LAST_PAYMENT_AMOUNT'];
-				} elseif ( isset( $metadata['TOTAL_SPENT'] ) ) {
-					$donation_amount = (int) $metadata['TOTAL_SPENT'];
-				}
-				if ( 0 < $donation_amount ) {
-					$events_to_add[] = [
-						'type'    => 'donation',
-						'context' => $newsletter_provider->service,
-						'value'   => [
-							'amount' => $donation_amount,
-						],
-					];
-				}
-			}
-		}
 
-		if ( ! empty( $events_to_add ) ) {
-			$api->save_reader_events( $client_id, $events_to_add );
+			// TODO: set donation status in the Reader Data store.
```

Comment:
That's been done in https://github.com/Automattic/newspack-plugin/pull/2539

Reply:
Ah, you're right—this file only checks subscription/donation status upon user login, but that's already handled by https://github.com/Automattic/newspack-plugin/pull/2539. 🙌

Reply:
Removed this file in 6c97c16